src/rendering/GLRenderer.hx

```diff
@@ -277,27 +278,60 @@ class GLRenderer
 		uvs.push(uvy + uvh);
 	}
 
-	public function drawTile(x:Float, y:Float, tileset: Tileset, id: Int): Void
+	public function drawTile(x:Float, y:Float, tileset:Tileset, tile:TileData): Void
 	{
 		setTexture(tileset.texture);
 
-		var tx = (id % tileset.tileColumns);
-		var ty = Math.floor(id / tileset.tileColumns);
+		var tx = (tile.idx % tileset.tileColumns);
+		var ty = Math.floor(tile.idx / tileset.tileColumns);
 		var tw = tileset.tileWidth;
 		var th = tileset.tileHeight;
 
-		positions.push(x);
-		positions.push(y);
-		positions.push(x + tw);
-		positions.push(y);
-		positions.push(x);
-		positions.push(y + th);
-		positions.push(x + tw);
-		positions.push(y);
-		positions.push(x);
-		positions.push(y + th);
-		positions.push(x + tw);
-		positions.push(y + th);
+		var topLeft = new Vector(-1, -1);
+		var topRight = new Vector(1, -1);
+		var botLeft = new Vector(-1, 1);
+		var botRight = new Vector(1, 1);
```

Comment:
Should probably define these and other temp Vectors as class members instead of locals